### PR TITLE
[FLINK-19393][docs-zh] Translate the 'SQL Hints' page of 'Table API & SQL' into Chinese

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

*Translate the 'SQL Hints' page of 'Table API & SQL' into Chinese*

## Brief change log

*Translate the 'SQL Hints' page of 'Table API & SQL' into Chinese*
  

## Verifying this change

*Translate the 'SQL Hints' page of 'Table API & SQL' into Chinese*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
@XBaith Could you help  me to  review this  PR?
Thank  you.

**Comment by RocMarshal** [comment]
@wuchong  Could you help me to review this PR?
Thank you.

**Comment by wuchong** [comment]
cc @fsk119 , could you help to review this PR?

File: docs/dev/table/sql/hints.zh.md [hunk 1]
```diff
@@ -25,34 +25,25 @@ under the License.
...
-Thus it is very suitable to use for the ad-hoc queries in interactive terminal, for example, in the SQL-CLI,
-you can specify to ignore the parse error for a CSV source just by adding a dynamic option `/*+ OPTIONS('csv.ignore-parse-errors'='true') */`.
-
-<b>Note:</b> Dynamic table options default is forbidden to use because it may change the semantics of the query.
-You need to set the config option `table.dynamic-table-options.enabled` to be `true` explicitly (default is false),
-See the <a href="{% link dev/table/config.zh.md %}">Configuration</a> for details on how to set up the config options.
-
-### Syntax
-In order to not break the SQL compatibility, we use the Oracle style SQL hint syntax:
+可以将 SQL 提示（hints）与 SQL 语句一起使用改变执行计划（execution plans）。本章介绍如何使用 SQL 提示增强各种方法。
```
Comment (fsk119) [inline docs/dev/table/sql/hints.zh.md:28]: I think it's better to not translate the word `hints`. After I search with key words `sql hints` in baidu.com, I get more results about auto complete.

I think it's better to translate to 

```
SQL hints 是和SQL语句一起使用来改变执行计划的。
```

File: docs/dev/table/sql/hints.zh.md [hunk 2]
```diff
@@ -25,34 +25,25 @@ under the License.
...
+
+SQL 提示一般可以用于以下：
+
+- Enforce planner：没有十分契合的 planner，所以实现提示让用户更好地控制执行是十分有意义的；
+- 增加元数据（或者统计信息）：如 "table index for scan" 和 "skew info of some shuffle keys" 的一些统计数据对于查询来说是动态的，用提示来配置它们会非常方便，因为我们从 planner 获得的计划元数据通常不那么准确；
```
Comment (fsk119) [inline docs/dev/table/sql/hints.zh.md:33]: Please also translate the "table index for scan" and "skew info of some shuffle keys".

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. I left some comments.

**Comment by RocMarshal** [comment]
@fsk119 Thank you for your comments. And I made some changes based on your suggestions. Please take a look.

**Review by fsk119 — APPROVED** [review]
Thanks for your update. LGTM.

**Comment by RocMarshal** [comment]
@wuchong I made some changes based on fsk119's suggestions. Please take a look.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**